### PR TITLE
iOS-2706 ObjectIconPicker & CoverPicker | Add service for handle uploading from any place

### PR DESCRIPTION
**Pull request by mgolovko**
Refactoring. Header still show uploading state.

https://github.com/anyproto/anytype-swift/assets/2483784/6639e9db-4e3d-4990-a261-5b4bc40baba2

### CODE REVIEW

File: Anytype/Sources/ServiceLayer/ObjectHeaderUploadingService/ObjectHeaderUploadingService.swift [hunk 1]
```diff
@@ -0,0 +1,112 @@
+import UIKit
+import Services
+import AnytypeCore
+import Combine
+
+protocol ObjectHeaderUploadingServiceProtocol: AnyObject {
+    
+    func coverUploadPublisher(objectId: String, spaceId: String) -> AnyPublisher<ObjectHeaderUpdate, Never>
```
Comment (mgolovko) [inline Anytype/Sources/ServiceLayer/ObjectHeaderUploadingService/ObjectHeaderUploadingService.swift:8]: Header handle uploading state from this method.
Old implementation used `actionHandler: @escaping (ObjectSettingsAction) -> Void`

File: Anytype/Sources/PresentationLayer/TextEditor/EditorPage/Views/Settings/Icon/ObjectIconPickerViewModel.swift [hunk 2]
```diff
@@ -24,22 +24,34 @@ final class ObjectIconPickerViewModel: ObservableObject, ObjectIconPickerViewMod
...
-        self.onIconAction = onIconAction
+        self.actionHandler = actionHandler
         subscription = document.syncPublisher.sink { [weak self] in
             self?.updateState()
         }
     }
     
+    
+    func setEmoji(_ emojiUnicode: String) {
+        actionHandler.handleIconAction(document: document, action: .setIcon(.emoji(emojiUnicode: emojiUnicode)))
```
Comment (mgolovko) [inline Anytype/Sources/PresentationLayer/TextEditor/EditorPage/Views/Settings/Icon/ObjectIconPickerViewModel.swift:42]: Now screen contains uploading logic. Not HeaderViewModel. Header just handle uploading state from ObjectHeaderUploadingServiceProtocol

File: Anytype/Sources/PresentationLayer/TextEditor/EditorPage/Views/Settings/Cover/ObjectCoverPickerViewModel.swift [hunk 3]
```diff
@@ -28,36 +27,67 @@ final class ObjectCoverPickerViewModel: ObservableObject {
...
-        self.onCoverAction = data.onCoverAction
     }
 }
 
 extension ObjectCoverPickerViewModel {
     
     func setColor(_ colorName: String) {
-        onCoverAction(.setCover(.color(colorName: colorName)))
+        Task {
+            try await objectHeaderUploadingService.handleCoverAction(
```
Comment (mgolovko) [inline Anytype/Sources/PresentationLayer/TextEditor/EditorPage/Views/Settings/Cover/ObjectCoverPickerViewModel.swift:46]: Now screen contains uploading logic. Not HeaderViewModel. Header just handle uploading state from ObjectHeaderUploadingServiceProtocol

File: Anytype/Sources/PresentationLayer/TextEditor/EditorPage/Views/Settings/Icon/Modules/ObjectIconPickerModuleAssembly.swift [hunk 4]
```diff
@@ -3,16 +3,11 @@ import UIKit
 import SwiftUI
 
 protocol ObjectIconPickerModuleAssemblyProtocol {
-    func make(
-        document: BaseDocumentGeneralProtocol,
-        onIconAction: @escaping (ObjectIconPickerAction) -> Void
```
Comment (mgolovko) [inline Anytype/Sources/PresentationLayer/TextEditor/EditorPage/Views/Settings/Icon/Modules/ObjectIconPickerModuleAssembly.swift:8]: Deleted - `onIconAction: @escaping (ObjectIconPickerAction) -> Void`
Now this screen contains icon uploading logic. No needed that this logic will be implemented in header view model

File: Anytype/Sources/ServiceLayer/ObjectHeaderUploadingService/ObjectHeaderUploadingService.swift [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    @Injected(\.unsplashService)
+    private var unsplashService: UnsplashServiceProtocol
+    
+    private var coverUploadSubject = PassthroughSubject<(objectId: String, spaceId: String, update: ObjectHeaderUpdate), Never>()
+    
+    func coverUploadPublisher(objectId: String, spaceId: String) -> AnyPublisher<ObjectHeaderUpdate, Never> {
+        coverUploadSubject.filter { $0.objectId == objectId && $0.spaceId == spaceId }.map { $0.update }.eraseToAnyPublisher()
+    }
+    
+    func handleCoverAction(
```
Comment (mgolovko) [inline Anytype/Sources/ServiceLayer/ObjectHeaderUploadingService/ObjectHeaderUploadingService.swift:34]: This code moved from header view model

**Review by joe-pusya — APPROVED** [review]

**Pull request merged**